packages/nextjs/hooks/scaffold-eth/useTargetNetwork.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from "react";
+import { useNetwork } from "wagmi";
+import scaffoldConfig from "~~/scaffold.config";
+import { useGlobalState } from "~~/services/store/store";
+import { ChainWithAttributes } from "~~/utils/scaffold-eth";
+import { NETWORKS_EXTRA_DATA } from "~~/utils/scaffold-eth";
+
+export function useTargetNetwork(): { targetNetwork: ChainWithAttributes } {
+  const { chain } = useNetwork();
+  const { targetNetwork, setTargetNetwork } = useGlobalState(({ targetNetwork, setTargetNetwork }) => ({
+    targetNetwork,
+    setTargetNetwork,
+  }));
```

Comment:
I think we could simply this 😅 : 

```ts
  const { targetNetwork, setTargetNetwork } = useGlobalState(state => ({
    targetNetwork: state.targetNetwork,
    setTargetNetwork: state.setTargetNetwork,
  }));
```

If we are concerned with referential equality maybe we could use pass `shallow` comparison to `useGlobalState`  or get each prop individually (which look fine too)

checkout this discussion : https://github.com/pmndrs/zustand/discussions/1916

Reply:
I split it up, looks cleaner like that